addons/web_editor/controllers/main.py

```diff
@@ -545,3 +547,43 @@ def subber(match):
             ('Content-type', 'image/svg+xml'),
             ('Cache-control', 'max-age=%s' % http.STATIC_CACHE_LONG),
         ])
+
+    @http.route(['/web_editor/media_library_url'], type='json', auth="user", website=True)
+    def media_library_url(self):
+        dbuuid = request.env['ir.config_parameter'].sudo().get_param('database.uuid')
+        endpoint = request.env['ir.config_parameter'].sudo().get_param('web_editor.media_library_endpoint', DEFAULT_LIBRARY_ENDPOINT)
+        return '%s?dbuuid=%s' % (endpoint, dbuuid)
+
+    @http.route('/web_editor/save_media', type='json', auth='user', methods=['POST'])
+    def save_media(self, media=[]):
+        """
+        Saves images from the media library as new attachments, making them
+        dynamic SVGs if needed.
+            media = [
+                {
+                    'media_url': 'https://...',
+                    'query': 'space separated search terms',
+                    'is_dynamic_svg': True/False,
+                }, ...
+            ]
+        """
+        attachments = []
+
+        for medium in media:
+            url = medium['media_url']
+            req = requests.get(url)
+            name = '_'.join([medium['query'], url.split('/')[-1]])
+            attachment = request.env['ir.attachment'].create({
+                'name': name,
+                'mimetype': req.headers['content-type'],
+                'datas': b64encode(req.content),
+                'public': True,
+                'res_model': 'ir.ui.view',
+                'res_id': 0,
+            })
+            if medium['is_dynamic_svg']:
+                # prefixing the name with id to ensure uniqueness of the url
+                attachment['url'] = '/web_editor/shape/external_shapes/%s_%s' % (attachment.id, name)
```

Comment:
And why not use our classic slug if you want to concat name and id...

Reply:
I need the url to be `/web_editor/shape/...` to hit the color-customizing controller

Reply:
external_shapes is a module that's a depends of web_editor ? not sure

no I mean: `name-id`  instead of `id_name` (standard slug in Odoo)

Reply:
external_shapes in not a module, this leverages the attachment fallback from the `/web_editor/shapes` route

Changed to name-id, didn't know we had a convention.